### DISCUSSION TITLE
Mistake during flashing, how to recover?

### DISCUSSION THREAD

**Discussion by Phonog** [arendst/Tasmota#17319] [Q&A] [answered]
Hi all,
I have flashed tasmota32.bin into a Sonoff Dual R3 using Esp-Flasher. Everything went well, but the processor keeps restarting after a few seconds of operation. In the meantime it outputs information on the serial port that I can read on my screen.
I then realized that according to the website, I should have chosen tasmota32.factory.bin. So I chose this file on Esp-flasher, and launched the flashing. Unfortunately I get :
Unexpected error: The firmware binary is invalid (magic byte=FF, should be E9)
What can I do now?

**Comment by barbudor** [comment, marked as answer]
ESP Flasher may not use the proper flashing parameter for Tasmota.
It's not because it can flash other firmware that it would work with Tasmota
The best would be to try Tasmota Web Flasher :  https://tasmota.github.io/install/
Just ask to flash "Tasmota" and it will automatically identify the appropriate variant for your firmware

**Reply by sfromis** [reply]
ESP-Flasher **does** not work for ESP32 when using the recommended `.factory` images

**Reply by sfromis** [reply]
ESP32 devices will often not be able to boot to Wifi via the power available from the USB/serial adapter. It may work better if you connect to mains power instead, after disconnecting the USB/serial adapter.

**Comment by Phonog** [comment]
I have tried the Tasmota Web Flasher.
Everything went ok. However, the Sonoff module would restart over and over, until I disconnected the serial converter and powered it directly from mains. Then I could have it work as expected.
Thank you!
